cs/client.py

```diff
@@ -219,11 +221,100 @@ def _request(self, command, json=True, opcode_name='command',
+                done = True
             else:
                 final_data = data
                 done = True
         return final_data
 
+    def _response_value(self, response, json=True):
+        """Parses the HTTP response as a the cloudstack value.
+
+        It throws an exception if the server didn't answer with a 200.
+        """
+        if json:
+            # XXX check for the content-type: application/json
+            try:
+                data = response.json()
+            except ValueError as e:
+                msg = "Make sure endpoint URL '%s' is correct." % self.endpoint
+                raise CloudStackException(
+                    "HTTP {0} response from CloudStack".format(
+                        response.status_code), response, "%s. " % str(e) + msg)
+
+            [key] = data.keys()
+            data = data[key]
+        else:
+            data = response.text
+
+        if response.status_code != 200:
+            raise CloudStackException(
+                "HTTP {0} response from CloudStack".format(
+                    response.status_code), response, data)
+
+        return data
+
+    def _jobresult(self, jobid, json=True, headers=None):
+        """Poll the async job result.
+
+        To be run via in a Thread, the result is put within
+        the result list which is a hack.
+        """
+        failures = 0
+
+        total_time = self.job_timeout or 2**30
+        remaining = timedelta(seconds=total_time)
+        endtime = datetime.now() + remaining
+
+        while remaining.total_seconds() > 0:
+            timeout = max(min(self.timeout, remaining.total_seconds()), 1)
+            try:
+                kind, params = self._prepare_request('queryAsyncJobResult',
+                                                     jobid=jobid)
+
+                transform(params)
+                params['signature'] = self._sign(params)
+
+                response = getattr(requests, self.method)(self.endpoint,
+                                                          headers=headers,
+                                                          timeout=timeout,
+                                                          verify=self.verify,
+                                                          cert=self.cert,
+                                                          **{kind: params})
+
+                j = self._response_value(response, json)
+
+                failures = 0
+                if j['jobstatus'] != PENDING:
+                    if j['jobresultcode'] or j['jobstatus'] != SUCCESS:
+                        raise CloudStackException("Job failure", response)
+
+                    if 'jobresult' not in j:
+                        raise CloudStackException("Unknown job result",
+                                                  response)
+
+                    return j['jobresult']
+
+            except CloudStackException as e:
+                raise e
```

Comment:
You can do `except CloudStackException: raise` to re-raise the last exception (same below)